test-suite/misc/deps-missing-vo/_CoqProject

```diff
@@ -0,0 +1,2 @@
+-Q . test
+A.v
```

Comment:
That's a bit weird, why are not `B.v` and `lib/D.v` here?

Is that a bug in coq_makefile? If so we should not workaround it in coqdep.

Note that for example Dune shouldn't have this problem, as it will see the generated modules so it will correctly generate them before calling `coqdep` because indeed that what the sound build rule is.

Reply:
If I list all the autogenerated .v files in `_CoqProject`, they will be autogenerated before make calls `coqdep`. Sorry, it was not clear from the test case but no, it is not weird, it is very much my main goal in this PR: to *not* have to eagerly generate all the autogenerated .v files, and to allow a user to quickly build the minimal dependencies of a .vo file as fast as possible so that the user can quickly edit the corresponding .v file interactively. Also, autogeneration of some .v files can fail and unnecessarily block the whole build system and prevent someone from building .vo files that don't even depend on the .v files whose autogeneration failed.
So, we only put the non-autogenerated .v files in the `_CoqProject` and let the build system figure out when it is really necessary to generate some .v files. Sorry, with just one non-autogenerated .v file in the test case, my point is not illustrated. I will add another one and invoke make to build only a specific .vo file and illustrate that it does not trigger the generation of any unnecessary .v file.

Reply:
> I will add another one to the test case and invoke make to build only a specific .vo file and illustrate that it does not trigger the generation of any unnecessary .v file.

To be more specific, the interesting test case here is one where the `.v` file may or may not be yet generated, `coqdep`'s output should not vary in this case, including when closed under several `-Q` options which could be also part of the current build DAG.

Reply:
the coqdep output is stable under the presence/absence of files (assuming no missing directories) as long as one makes sure that the -Q options are such that no two physical directories correspond to the same logical path, which I think is a sensible invariant in most projects.
More generally, to use the feature in this PR, the user is supposed to ensure the following:
Suppose a file `/d1/d2/d3..dn/f.v` is missing and directories `dk/../dn` also don't exist. 
The user needs to make sure their `-Q` directives are such that only one logical path corresponds to the physical path `/d1/d2/..dk-1`. It is also possible to complicate the current implementation and be able to detect and fail in case of such ambiguities.

Reply:
the case here https://github.com/aa755/coq/tree/coqdep-master/test-suite/misc/deps-missing-vo has several -Q options

Reply:
> In Coq + dune this is doable, tho there you have another workaround for your cold-start problem if you can split your dev into different `theories`.

AFAIK, dune just invokes coqdep for computing dependencies. So I am not sure how dune is relevant here.

Reply:
> AFAIK, dune just invokes coqdep for computing dependencies. So I am not sure how dune is relevant here.

Dune has knowledge about the global module structure including generated targets, so it can perform the mapping itself without looking at filesystem, as mentioned above it does for OCaml using `ocamldeps` `-modules` option, which is stable under FS changes, thus safe in an incremental context.